compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -149,6 +149,9 @@ pub fn time_trace_profiler_finish(file_name: &Path) {
 // Though note that Rust can also be build with an external precompiled version of LLVM
 // which might lead to failures if the oldest tested / supported LLVM version
 // doesn't yet support the relevant intrinsics
+//
+// Note: The first feature in the list that is returned is the mapping to the feature that is
+// provided from the `s` parameter.
 pub fn to_llvm_features<'a>(sess: &Session, s: &'a str) -> SmallVec<[&'a str; 2]> {
```

Comment:
I think it's a mistake to make this function do too much. Could the logic for implied/tied features be moved to a separate function and handled separately?

Reply:
Yeah, I agree with that and was thinking the same while making the change. This was already handling the implied features before this change, so I just added `neon` and `fp-armv8` to it to fix that issue. The fact it's used in multiple places is why I created a different function for if they should be disabled together.

It was done this way to fix those bugs, however, long term I think this should be a graph structure to correctly show the relationships between the different features.

Reply:
Perhaps it would be clearer to make this function return a struct instead:
```rust
struct LlvmFeatures<'a> {
    /// Features that should be enabled/disabled.
    features: smallvec![&'a str],

    /// Features that are implicitly enabled when the feature is enabled.
    dependencies: smallvec![&'a str],
}
```